driver/modern_bpf/helpers/store/auxmap_store_params.h

```diff
@@ -1137,6 +1128,30 @@ static __always_inline void auxmap__store_iovec_data_param(struct auxiliary_map
 	push__param_len(auxmap->data, &auxmap->lengths_pos, total_size_to_read);
 }
 
+/**
+ * @brief Store a message extracted from iovec structs.
+ *
+ * @param auxmap pointer to the auxmap in which we are storing the param.
+ * @param msghdr_pointer pointer to `user_msghdr` struct.
+ * @param len_to_read imposed snaplen.
+ */
+static __always_inline void auxmap__store_msghdr_iovec_data_param(struct auxiliary_map *auxmap, unsigned long msghdr_pointer, unsigned long len_to_read)
+{
+	/* Read the usr_msghdr struct into the stack, if we fail,
+	 * we return an empty param.
+	 */
+	struct user_msghdr msghdr = {0};
+	if(bpf_probe_read_user((void *)&msghdr, bpf_core_type_size(struct user_msghdr), (void *)msghdr_pointer))
+	{
+		push__param_len(auxmap->data, &auxmap->lengths_pos, 0);
+		return;
+	}
+
+	u32 iov_cnt = msghdr.msg_iovlen;
+
+	auxmap__store_iovec_data_param(auxmap, (unsigned long)msghdr.msg_iov, iov_cnt, len_to_read);
+}
```

Comment:
Note for the reviewer: the message header structure `msghdr` wraps a pointer to an array of `iovec`. Let's just isolate the handling of `iovec`, so that it can be reused directly in `readv` fillers.